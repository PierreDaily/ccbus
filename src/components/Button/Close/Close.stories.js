import CloseBtn from "./index.vue";

// More on default export: https://storybook.js.org/docs/vue/writing-stories/introduction#default-export
export default {
  title: "CloseBtn",
  component: CloseBtn,
  // More on argTypes: https://storybook.js.org/docs/vue/api/argtypes
  argTypes: {},
};

// More on component templates: https://storybook.js.org/docs/vue/writing-stories/introduction#using-args
const Template = (args) => ({
  // Components used in your story `template` are defined in the `components` object
  components: { CloseBtn },
  // The story's `args` need to be mapped into the template through the `setup()` method
  setup() {
    return { args };
  },
  // And then the `args` are bound to your component with `v-bind="args"`
  template: '<close-btn v-bind="args" />',
});

export const WithURL = Template.bind({});
// More on args: https://storybook.js.org/docs/vue/writing-stories/args

WithURL.args = {
  to: "http://mywebsite.com",
};
