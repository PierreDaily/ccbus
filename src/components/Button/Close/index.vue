<template>
    <router-link :to="to">
        <img class="h-10 w-10" :src="closeIcon" />
    </router-link>
</template>

<script lang="ts">
import { defineComponent } from 'vue';
import closeIcon from "@/assets/icons/close.svg";
export default defineComponent({
    name: 'CloseBtn',
    props: {
        to: {
            type: String,
            required: true
        }
    },
    setup() {
        return {
            closeIcon,
        };
    },
})
</script>