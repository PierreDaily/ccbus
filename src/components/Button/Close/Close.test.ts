import { mount, shallowMount } from "@vue/test-utils";
import Close from "./index.vue";

test("Mount component", async () => {
  // arrange
  expect(Close).toBeTruthy();

  const wrapper = mount(Close, {
    props: {
      to: "/test",
    },
  });

  // assert
  expect(wrapper.html()).toMatchSnapshot();
});

test("Mount component", async () => {
    // arrange
    const URL = "/test"
    expect(Close).toBeTruthy();
  
    const wrapper = mount(Close, {
      props: {
        to: URL,
      },
    });
  
    // assert
    expect(wrapper.find("router-link").exists()).toBe(true)
    expect(wrapper.find("router-link").attributes("to")).toBe(URL)
  });
