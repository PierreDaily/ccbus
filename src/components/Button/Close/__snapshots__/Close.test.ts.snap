// Vitest Snapshot v1

exports[`Mount component 1`] = `"<router-link to=\\"/test\\" title=\\"close\\"><img class=\\"h-10 w-10\\" src=\\"/src/assets/icons/close.svg\\" /></router-link>"`;
